proto-public/pbcatalog/v1alpha1/service_endpoints.proto

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+syntax = "proto3";
+
+package hashicorp.consul.catalog.v1alpha1;
+
+import "pbcatalog/v1alpha1/health.proto";
+import "pbcatalog/v1alpha1/protocol.proto";
+import "pbcatalog/v1alpha1/workload.proto";
+import "pbresource/resource.proto";
+
+message ServiceEndpoints {
+  repeated Endpoint endpoints = 1;
+}
+
+message Endpoint {
+  // target_ref is the reference to the resource
+  // for this endpoint endpoint. This currently must be a workload.
+  hashicorp.consul.resource.ID target_ref = 1;
+
+  // addresses is the list of addresses for this endpoint.
+  // This has the same structure as the workload addresses.
+  repeated WorkloadAddress addresses = 2;
+
+  // ports is the map of ports for this endpoint.
+  // This has the same structure as the workload ports.
+  map<string, EndpointPort> ports = 3;
+
+  // health_status is the aggregated health status of this endpoint.
+  Health health_status = 4;
+}
+
+message EndpointPort {
+  // virtual_port is the port that could only be used when transparent
+  // proxy is used alongside a virtual IP or a virtual DNS address.
+  // This value is ignored in other cases. Whether or not using transparent
+  // proxy, this value is optional.
+  uint32 virtual_port = 1;
+
+  // phsical_port is the real port that the workload has bound to.
+  uint32 physical_port = 2;
```

Comment:
Add target port name to make proxy config doable and set the ALPN appropriately.